File: CodingSamples/Java/Runtime/ReflectionTest1/Program2.java
class Program {

	private static void present(Object info) {
		Class<?> c = info.getClass();
		System.out.printf("<%s>%n", c.getName());
		for(var f : c.getDeclaredFields()){
			System.out.printf("  <%s>", f.getName());
			f.setAccessible(true);
			try{
				System.out.print(f.get(info));
			}catch(Exception e){
				System.out.print("Error");
			}
			System.out.printf("</%s>%n", f.getName());
		}
		System.out.printf("</%s>%n", c.getName());
		System.out.println();
	}

	public static void main(String[] args) {
		String uid = System.getProperty("user.name");
		present(new Customer(uid, 45000, 3));	
		present(new Item("cpu", "intel"));
		present(new Interval(3, 45));
	}
}

File: CodingSamples/DotNet/Runtime/ReflectionTest1/DemoApp/Program2.cs
using System.Reflection;

class Program
{
    private static void Present(object info)
    {
        Type t = info.GetType();
        Console.WriteLine("<{0}>", t.Name);
        foreach(PropertyInfo p in t.GetProperties())
            Console.WriteLine("  <{0}>{1}</{0}>", p.Name, p.GetValue(info));
        Console.WriteLine("</{0}>", t.Name);
        Console.WriteLine();
    }

    public static void Main(string[] args)
    {
        Present(new Item("cpu", "intel"));
        Present(new Customer("Jack", 23000, 4));
    }
}
